action.yml

```diff
@@ -37,23 +37,34 @@ inputs:
     description: consumer secret generated for the entire app
     required: false
 
+# If you want to post to Mastodon, these values are required
+  mastdodon_deploy:
+    description: Boolean to deploy to Mastodon
+    required: false
+  mastodon_access_token:
+    description: API key generated for the user account to tweet
+    required: false
+  nastodon_api_base_url:
```

Comment:
Should this perhaps have a default to mastodon.social's space?

Reply:
I don't think there really is any good default.  It's not centralized like Twitter.  I admit that I'm not Mastodon expert but I have seen hundreds of Mastodon instances cropping up in the course of the great exodus.

Reply:
That's fair! I think we are possibly setting the user up to make a mistake - e.g., assume a default and want to deploy but then they don't have all the metadata, and it just posts to twitter/slack. How about we add a check (before any of that happens) to protect them - the entire thing should fail (before doing any posts) if they intended to deploy to mastodon but forgot an envar. E.g., something like:

```python
if args.deploy_mastodon and not mastodon_client:
    sys.exit("Mastodon deploy requested, but environment variables are missing.")
elif args.deploy_mastodon and mastodon_client:
    <what you currently have>
```
We would probably want this pattern for the others (if we don't have it yet) but that's out of scope for this PR if you don't want to add it. In the meantime, let's do these checks earlier so we don't have partial posting. After that + the test run I think we should be good to go!

Reply:
I think the check that you are suggesting already happens in `get_mastodon_client()` at lines 127-129, no?

Reply:
doh, missed that! Yes you are spot on. So the remaining item is to use this branch in a repo somewhere to demonstrate that it works (and we can preview the toots to make sure they look okay - e.g., I remember for Twitter/Slack we had to do some work to get the images / links to expand).

Reply:
It's still a bit tricky for me to trigger this in my own branch but I'm working on it and will come back to this soon.

Reply:
Can you not update the US-RSE workflow to just use that? :thinking: I'm happy to give you a branch here too, but that would be equivalent to the first. Let me know your preference!

Reply:
I don't have privileges on the US-RSE repository.
I fixed up the workflow to run in my own repository, see: https://github.com/crd477/jobs-updater/actions/runs/3796882109/jobs/6457370371 and the result here: https://mastodon.sdf.org/@crd/109593600164877621

Reply:
Oh, on a related note, I realized that the action on the US-RSE repo is using `rseng/jobs-updater@add/multiple-metadata-fields`.  Do you want me to resubmit my mastodon branch against that or do you want to merge it into `rseng/jobs-updater@main` and then I deal with merge conflicts?
Sorry, I should've considered this earlier...

Reply:
Oh gosh, good catch! I just completely forgot about that. We can do whatever is easiest for you - I do remember we needed that fix/change, so we don't want to eliminate it. I can either squash and merge that or something else. What would you prefer?

Reply:
If you feel that it is appropriate, how about if you merge `add/multiple-metadata-fields` into `main` and then I will deal with the merge conflicts after that.  Does that sound reasonable?  If you actually prefer to maintain that as a separate branch, then I can also resubmit the PR against that instead.

There will be conflicts either way but it looks like they should not be too awful regardless.

Reply:
All set! And when your work goes in and the branch over at US-RSE is updated I can delete that old one. It should have been merged like a year ago, sorry for the oversight!

Reply:
OK, I think this is ready to merge.  Please check out the last action on my fork:
https://github.com/crd477/jobs-updater/actions/runs/3797373182
and the resulting toots:
https://mastodon.sdf.org/@crd

Let me know if you think it looks OK.  Once it goes into main here, I'll see about fixing the usrse actions separately.
Thanks!

Reply:
I don't think that run is updated with commits here? I still see the warning about set-output, checkout@v2, and then the incorrect input param.  Is that the right link?

Also - tests need to pass here - so if it's trying to use mastodon (but cannot) we shouldn't be testing it here (because we cannot!)

Reply:
How did you test Slack and Twitter from here?  Can a similar thing be done for Mastodon?

Reply:
We only have a slack webhook defined (I'm not there anymore but you could check the US-RSE slack) and note that the workflow is just explicitly to check slack (no twitter). For Twitter we had to do the same we are doing here - test somewhere else with an actual token before merging here.